db/migration/V021__create_assessments.sql

```diff
@@ -0,0 +1,137 @@
+CREATE TABLE question_types (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  title TEXT NOT NULL,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE assessment_submission_states (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  title TEXT NOT NULL,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE program_participant_roles (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  title TEXT NOT NULL,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE curriculum_assessments (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  title TEXT NOT NULL,
+  description TEXT,
+  max_score INT NOT NULL,
+  max_num_submissions INT NOT NULL,
+  time_limit INT,
+  curriculum_id BIGINT NOT NULL,
+  activity_id BIGINT NOT NULL,
+  principal_id BIGINT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX curriculum_assessments_curriculum_id (curriculum_id),
+  FOREIGN KEY (curriculum_id) REFERENCES curriculums(id),
+  INDEX curriculum_assessments_activity_id (activity_id),
+  FOREIGN KEY (activity_id) REFERENCES activities(id),
+  INDEX curriculum_assessments_principal_id (principal_id),
+  FOREIGN KEY (principal_id) REFERENCES principals(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE program_assessments (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  program_id BIGINT NOT NULL,
+  assessment_id BIGINT NOT NULL,
+  available_after TIMESTAMP NOT NULL,
+  due_date TIMESTAMP NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  INDEX program_assessments_program_id (program_id),
+  FOREIGN KEY (program_id) REFERENCES programs(id),
+  INDEX program_assessments_assessment_id (assessment_id),
+  FOREIGN KEY (assessment_id) REFERENCES curriculum_assessments(id),
+  PRIMARY KEY (id)
+);
```

Comment:
Let's also add in a unique key here after we define the primary key.

```sql
UNIQUE KEY `program_and_assessment` (`program_id`,`assessment_id`)
```

Reply:
added UNIQUE KEY `program_and_assessment` (`program_id`,`assessment_id`)